src/app/content/types.ts

```diff
@@ -22,10 +22,7 @@ export interface Params {
 export interface State {
   tocOpen: boolean | null;
   params: Params | null;
-  loading: {
-    book?: Params['book']
-    page?: Params['page']
-  };
+  loading: Partial<Params>;
```

Comment:
I think this might be better? If its just `Params | null` it implies that if its truthy then both `book` and `page` are loading. I guess ts didnt check it since the spread operator was used in reducer while setting the `loading`

Reply:
yes i think this is good